### PR TITLE
Add Visual Test for Snow Theme and add visual tests to PR execution

### PR DESCRIPTION
**Pull request by unlikelyzero**
Closes: #5478 and covers

### Describe your changes:
Visual Testing to CI
- Adds visual testing to a new circle CI check
- Adds debuggability for visual test failure
- Caches the percy cli chrome install so that builds are faster
- Removes Idle Waits in Visual Suites to run as fast as possible
- Refactors visual suites to enable them to run with App Actions and Plugin Fixtures
- Removes github action for visual tests
- Runs visual tests on every merge
- Marked unstable visual tests where necessary
- Updates `percy/cli` to `1.6.4`

Snow Theme
- Updates playwright visual project to include parameterized chrome and chrome snow modes of operation

Drive-bys
- removes references to @frramework tags

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/nasa/openmct/blob/master/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/nasa/openmct/pulls) for the same update/change?
* [x] Is this change backwards compatible? For example, developers won't need to change how they are calling the API or how they've extended core plugins such as Tables or Plots.

### Author Checklist

* [x] Changes address original issue?
* [x] Tests included and/or updated with changes?
* [x] Command line build passes?
* [x] Has this been smoke tested?
* [x] Testing instructions included in associated issue OR is this a dependency/testcase change?

### Reviewer Checklist

* [ ] Changes appear to address issue?
* [ ] Changes appear not to be breaking changes?
* [ ] Appropriate unit tests included?
* [ ] Code style and in-line documentation are appropriate?
* [ ] Commit messages meet standards?
* [ ] Has associated issue been labelled unverified? (only applicable if this PR closes the issue)
* [ ] Has associated issue been labelled bug? (only applicable if this PR is for a bug fix)

### CODE REVIEW

File: e2e/playwright-visual.config.js [hunk 1]
```diff
@@ -2,30 +2,49 @@
...
-/** @type {import('@playwright/test').PlaywrightTestConfig} */
+/** @type {import('@playwright/test').PlaywrightTestConfig<{ theme: string }>} */
 const config = {
     retries: 0, // visual tests should never retry due to snapshot comparison errors
     testDir: 'tests/visual',
-    timeout: 90 * 1000,
-    workers: 1, // visual tests should never run in parallel due to test pollution
+    testMatch: '**/*.visual.spec.js', // only run visual tests
+    timeout: 60 * 1000,
+    workers: 2, // TODO See if this works
```
Comment (ozyx) [inline e2e/playwright-visual.config.js:11]: I think we can say it works now? :)

File: e2e/playwright-visual.config.js [hunk 2]
```diff
@@ -2,30 +2,49 @@
...
     },
+    projects: [
+        {
+            name: 'chrome',
+            use: {
+                browserName: 'chromium'
+            }
+        },
+        {
+            name: 'snow-theme-chrome',
```
Comment (ozyx) [inline e2e/playwright-visual.config.js:34]: These project names can be human readable (including spaces etc).

Maybe we can update with something like

```
{
    name: 'Chrome (Default Theme)',
    use: {
        browserName: 'chromium'
    }
},
{
    name: 'Chrome (Snow Theme)',
    use: {
        browserName: 'chromium',
        theme: 'snow'
    }
}
```

File: e2e/tests/visual/default.visual.spec.js [hunk 3]
```diff
@@ -64,53 +58,30 @@ test.describe('Visual - Default', () => {
         await versionInformationLocator.evaluate(node => node.innerHTML = '<li>Version: visual-snapshot</li> <li>Build Date: Mon Nov 15 2021 08:07:51 GMT-0800 (Pacific Standard Time)</li> <li>Revision: 93049cdbc6c047697ca204893db9603b864b8c9f</li> <li>Branch: master</li>');
 
         // Take a snapshot of the About modal
-        await page.waitForTimeout(VISUAL_GRACE_PERIOD);
-        await percySnapshot(page, 'About');
+        await percySnapshot(page, 'About' + theme);
```
Comment (ozyx) [inline e2e/tests/visual/default.visual.spec.js:61]:
```suggestion
        await percySnapshot(page, `About (theme: '${theme}')`);
```

File: e2e/tests/visual/default.visual.spec.js [hunk 4]
```diff
@@ -64,53 +58,30 @@ test.describe('Visual - Default', () => {
...
+    test('Visual - Default Condition Set', async ({ page, theme }) => {
 
-        // Click text=OK
-        await page.click('text=OK');
+        await createDomainObjectWithDefaults(page, 'Condition Set');
 
         // Take a snapshot of the newly created Condition Set object
-        await page.waitForTimeout(VISUAL_GRACE_PERIOD);
-        await percySnapshot(page, 'Default Condition Set');
+        await percySnapshot(page, 'Default Condition Set' + theme);
```
Comment (ozyx) [inline e2e/tests/visual/default.visual.spec.js:69]:
```suggestion
        await percySnapshot(page, `Default Condition Set (theme: '${theme}')`);
```

File: e2e/tests/visual/default.visual.spec.js [hunk 5]
```diff
@@ -64,53 +58,30 @@ test.describe('Visual - Default', () => {
...
-        await page.click('text=Condition Widget');
 
-        // Click text=OK
-        await page.click('text=OK');
+        await createDomainObjectWithDefaults(page, 'Condition Widget');
 
         // Take a snapshot of the newly created Condition Widget object
-        await page.waitForTimeout(VISUAL_GRACE_PERIOD);
-        await percySnapshot(page, 'Default Condition Widget');
+        await percySnapshot(page, 'Default Condition Widget' + theme);
```
Comment (ozyx) [inline e2e/tests/visual/default.visual.spec.js:81]:
```suggestion
        await percySnapshot(page, `Default Condition Widget (theme: '${theme}')`);
```

File: e2e/tests/visual/default.visual.spec.js [hunk 6]
```diff
@@ -123,16 +94,14 @@ test.describe('Visual - Default', () => {
         await page.locator('input[type="text"]').first().fill(startDate.toString());
 
         //  verify no error msg
-        await page.waitForTimeout(VISUAL_GRACE_PERIOD);
-        await percySnapshot(page, 'Default Time conductor');
+        await percySnapshot(page, 'Default Time conductor' + theme);
```
Comment (ozyx) [inline e2e/tests/visual/default.visual.spec.js:97]:
```suggestion
        await percySnapshot(page, `Default Time conductor (theme: '${theme}')`);
```

File: e2e/tests/visual/default.visual.spec.js [hunk 7]
```diff
@@ -123,16 +94,14 @@ test.describe('Visual - Default', () => {
...
 
         startDate = (year + 1) + startDate.substring(4);
         await page.locator('input[type="text"]').first().fill(startDate.toString());
         await page.locator('input[type="text"]').nth(1).click();
 
         //  verify error msg for start time (unable to capture snapshot of popup)
-        await page.waitForTimeout(VISUAL_GRACE_PERIOD);
-        await percySnapshot(page, 'Start time error');
+        await percySnapshot(page, 'Start time error' + theme);
```
Comment (ozyx) [inline e2e/tests/visual/default.visual.spec.js:104]:
```suggestion
        await percySnapshot(page, `Start time error (theme: '${theme}')`);
```

File: e2e/tests/visual/default.visual.spec.js [hunk 8]
```diff
@@ -143,79 +112,50 @@ test.describe('Visual - Default', () => {
         await page.locator('input[type="text"]').first().click();
 
         //  verify error msg for end time (unable to capture snapshot of popup)
-        await page.waitForTimeout(VISUAL_GRACE_PERIOD);
-        await percySnapshot(page, 'End time error');
+        await percySnapshot(page, 'End time error' + theme);
```
Comment (ozyx) [inline e2e/tests/visual/default.visual.spec.js:115]:
```suggestion
        await percySnapshot(page, `End time error (theme: '${theme}')`);
```

File: e2e/tests/visual/default.visual.spec.js [hunk 9]
```diff
@@ -143,79 +112,50 @@ test.describe('Visual - Default', () => {
...
+    test('Visual - Sine Wave Generator Form', async ({ page, theme }) => {
         //Click the Create button
         await page.click('button:has-text("Create")');
 
         // Click text=Sine Wave Generator
         await page.click('text=Sine Wave Generator');
 
-        await page.waitForTimeout(VISUAL_GRACE_PERIOD);
-        await percySnapshot(page, 'Default Sine Wave Generator Form');
+        await percySnapshot(page, 'Default Sine Wave Generator Form' + theme);
```
Comment (ozyx) [inline e2e/tests/visual/default.visual.spec.js:125]:
```suggestion
        await percySnapshot(page, `Default Sine Wave Generator Form (theme: '${theme}')`);
```

File: e2e/tests/visual/default.visual.spec.js [hunk 10]
```diff
@@ -143,79 +112,50 @@ test.describe('Visual - Default', () => {
...
 
         await page.locator('.field.control.l-input-sm input').first().click();
         await page.locator('.field.control.l-input-sm input').first().fill('');
 
         // Validate red x mark
-        await page.waitForTimeout(VISUAL_GRACE_PERIOD);
-        await percySnapshot(page, 'removed amplitude property value');
+        await percySnapshot(page, 'removed amplitude property value' + theme);
```
Comment (ozyx) [inline e2e/tests/visual/default.visual.spec.js:131]:
```suggestion
        await percySnapshot(page, `removed amplitude property value (theme: '${theme}')`);
```

File: e2e/tests/visual/default.visual.spec.js [hunk 11]
```diff
@@ -143,79 +112,50 @@ test.describe('Visual - Default', () => {
...
-        await page.click('button:has-text("Create")');
-
-        //NOTE Something other than example imagery
-        await page.click('text=Timer');
-
-        // Click text=OK
-        await page.click('text=OK');
         await page.locator('.c-message-banner__message').hover({ trial: true });
-        await percySnapshot(page, 'Banner message shown');
+        await percySnapshot(page, 'Banner message shown' + theme);
```
Comment (ozyx) [inline e2e/tests/visual/default.visual.spec.js:138]:
```suggestion
        await percySnapshot(page, `Banner message shown (theme: '${theme}')`);
```

File: e2e/tests/visual/default.visual.spec.js [hunk 12]
```diff
@@ -143,79 +112,50 @@ test.describe('Visual - Default', () => {
...
 
         //Wait until Save Banner is gone
         await page.waitForSelector('.c-message-banner__message', { state: 'detached'});
-        await percySnapshot(page, 'Banner message gone');
+        await percySnapshot(page, 'Banner message gone' + theme);
```
Comment (ozyx) [inline e2e/tests/visual/default.visual.spec.js:142]:
```suggestion
        await percySnapshot(page, `Banner message gone (theme: '${theme}')`);
```

File: e2e/tests/visual/default.visual.spec.js [hunk 13]
```diff
@@ -143,79 +112,50 @@ test.describe('Visual - Default', () => {
...
-        await page.goto('/', { waitUntil: 'networkidle' });
-
+    test('Visual - Display Layout Icon is correct', async ({ page, theme }) => {
         //Click the Create button
         await page.click('button:has-text("Create")');
 
         //Hover on Display Layout option.
         await page.locator('text=Display Layout').hover();
-        await percySnapshot(page, 'Display Layout Create Menu');
+        await percySnapshot(page, 'Display Layout Create Menu' + theme);
```
Comment (ozyx) [inline e2e/tests/visual/default.visual.spec.js:151]:
```suggestion
        await percySnapshot(page, `Display Layout Create Menu (theme: '${theme}')`);
```

File: e2e/tests/visual/default.visual.spec.js [hunk 14]
```diff
@@ -143,79 +112,50 @@ test.describe('Visual - Default', () => {
...
-
-        await page.click('text=OK');
+    test('Visual - Default Gauge is correct', async ({ page, theme }) => {
+        await createDomainObjectWithDefaults(page, 'Gauge');
 
         // Take a snapshot of the newly created Gauge object
-        await page.waitForTimeout(VISUAL_GRACE_PERIOD);
-        await percySnapshot(page, 'Default Gauge');
-
+        await percySnapshot(page, 'Default Gauge' + theme);
```
Comment (ozyx) [inline e2e/tests/visual/default.visual.spec.js:159]:
```suggestion
        await percySnapshot(page, `Default Gauge (theme: '${theme}')`);
```

File: e2e/tests/visual/search.visual.spec.js [hunk 15]
```diff
@@ -24,81 +24,52 @@
...
+
         await page.locator('[aria-label="OpenMCT Search"] [aria-label="Search Input"]').fill('Cl');
-        // Click text=Unnamed Clock
+
         await Promise.all([
             page.waitForNavigation(),
             page.locator('text=Unnamed Clock').click()
         ]);
-        await percySnapshot(page, 'Clicking on search results should navigate to them if not editing');
+        await percySnapshot(page, 'Clicking on search results should navigate to them if not editing' + theme);
```
Comment (ozyx) [inline e2e/tests/visual/search.visual.spec.js:71]:
```suggestion
        await percySnapshot(page, `Clicking on search results should navigate to them if not editing (theme: '${theme}')`);
```

File: e2e/pluginFixtures.js [hunk 16]
```diff
@@ -89,6 +90,13 @@ async function getOrCreateDomainObject(page, options) {
  */
 const objectCreateOptions = null;
 
+/**
+ * If provided, this will run start the Open MCT application in Snow Theme. This should be referenced in a Project config
+ * and will initialize Open MCT with the Snow theme which is used in some other Open MCT projects.
+ * @type {'snow' | null}
```
Comment (ozyx) [inline e2e/pluginFixtures.js:96]:
```suggestion
 * @type {'snow' | 'espresso'}
```

**Review by ozyx — CHANGES_REQUESTED** [review]
Suggesting some minor changes in regards to defaults / visual test name clarity but overall looks great!

File: e2e/playwright-visual.config.js (hunk 1 above)
Comment (davetsay) [inline e2e/playwright-visual.config.js:11]: Are you still seeing if this works or can this be removed?

**Review by davetsay — CHANGES_REQUESTED** [review]
LGTM. This will be useful.

I just have one question to be addressed and it looks good otherwise.

File: e2e/playwright-visual.config.js (hunk 1 above)
Comment (unlikelyzero) [reply e2e/playwright-visual.config.js:11]: This works and can be kept :) This should double execution speed

File: e2e/playwright-visual.config.js (hunk 2 above)
Comment (unlikelyzero) [reply e2e/playwright-visual.config.js:34]: that would be awesome but then our users would have to pass that on the command line :(

File: e2e/playwright-visual.config.js (hunk 1 above)
Comment (unlikelyzero) [reply e2e/playwright-visual.config.js:11]: works!

File: e2e/pluginFixtures.js [hunk 17]
```diff
@@ -89,6 +90,13 @@ async function getOrCreateDomainObject(page, options) {
  */
 const objectCreateOptions = null;
 
+/**
+ * If provided, this will run start the Open MCT application in Snow Theme. This should be referenced in a Project config
```
Comment (ozyx) [inline e2e/pluginFixtures.js:94]: Need to fix the comment here

**Review by davetsay — APPROVED** [review]

**Review by ozyx — APPROVED** [review]
LGTM

**Pull request merged**